test/geoloco/utilities.py

```diff
@@ -0,0 +1,81 @@
+""""
+Common utility functions used by the geoloco regression tests.
+"""
+import os
+from pyhdf.SD import SD, SDC
+
+from harmony import Client, Request
+from harmony.harmony import ProcessingFailedException
+
+
+def submit_and_download(harmony_client: Client, request: Request,
+                        file_indicator: str) -> str:
+    """ Submit a Harmony request via a `harmony-py` client. Wait for the
+        Harmony job to finish, then download the results to the specified file
+        path.
+
+    """
+    downloaded_filenames = []
+    output_filename = None
+
+    try:
+        job_id = harmony_client.submit(request)
+
+        for filename in [file_future.result()
+                         for file_future
+                         in harmony_client.download_all(job_id,
+                                                        overwrite=True)]:
+
+            print(f'Downloaded: {filename}')
+            downloaded_filenames.extend([filename])
+
+        for filename in downloaded_filenames:
+            if file_indicator in filename:
+                output_filename = filename
+                print(f'Saved output to: {output_filename}')
+
+
+    except ProcessingFailedException as exception:
+        print_error('Harmony request failed to complete successfully.')
+        raise exception
+
+    return output_filename
+
+
+def get_dim_sizes(file: str) -> list[int]:
+    ydim = None
+    xdim = None
+    file_SD = SD(file, SDC.READ)
+    datasets = file_SD.datasets()
+    if len(datasets) > 0:
+        dataset_names = datasets.keys()
+        for name in dataset_names:
+            dataset = file_SD.select(name)
+            dims = dataset.dimensions(full=1)
+            ydim = list(dims.values())[0][0]
+            xdim = list(dims.values())[1][0]
+        return [ydim, xdim]
+
+
+def remove_results_files() -> None:
```

Comment:
@cbrownstevenson is this testing with `run_notebook.sh` or in Bamboo? And what would be hanging? I think the entire test suite takes about an hour.

Reply:
I actually tried to run it both ways, with run_notebook in SIT and UAT as well as just as a notebook against both also. You can look at the logs in cloudwatch to see the exact commands that took longer than an hour.    I just went back, and found a single request taking over 2 hours. Here's that https://github.com/nasa/harmony-regression-tests/pull/80#discussion_r1414233071

Reply:
I'm locked out of cloudtamer right now, but I'm able to look at the logs from harmony 	for request 7edbc884-6f64-4be2-a34f-c974f5724c40 in SIT.  My best guess is that it timed out in two hours, and then succeeded in 6 min on it's retry.  

I think the same thing happened with  0c4890d5-ecee-42ba-bf50-202ef9d06444 request in UAT.
start: 12/1/2023, 11:44:20 AM 
Finish: 12/1/2023, 2:20:38 PM

I'm looking at the logs and there are lots of ERROR level logs, but it seems like it might actually be doing what you want? 

```
"{\"message\": \" IS geolocated -- SensorZenith\\n\", \"user\": \"matthewsavoie\", \"requestId\": \"0c4890d5-ecee-42ba-bf50-202ef9d06444\", \"timestamp\": \"2023-12-01T21:19:59.901458Z\", \"level\": \"ERROR\", \"application\": \"harmony_python_interface/adapter.py\"}\r\n{\"message\": \" IS geolocated -- SensorAzimuth\\n\", \"user\": \"matthewsavoie\", \"requestId\": \"0c4890d5-ecee-42ba-bf50-202ef9d06444\", \"timestamp\": \"2023-12-01T21:19:59.901543Z\", \"level\": \"ERROR\", \"application\": \"harmony_python_interface/adapter.py\"}\r\n",
```

Reply:
@flamingbear hm... It wasn't 2 hours long for me. It was an hour long it SIT. What would be a good testing time length? The actually software is a c++ based executable.

Reply:
I am not currently able to look at logs. I need to submit a NAMS request.

Reply:
I think even an hour is pretty long.  If there was a way to trim it down to 15min that would be pretty good. There isn't a real defined limit, except the two hour timeout built into harmony, but you have to wait for the regressions on every deployment.